gateway/metrics/exporter.go

```diff
@@ -85,18 +86,24 @@ func (e *Exporter) StartServiceWatcher(endpointURL url.URL, metricsOptions Metri
 		},
 	}
 
+	endpointURL.Path = path.Join(endpointURL.Path, "system/functions")
+
+	// The request here can be cached safely as long as the body is nil.
+	serviceReq, err := http.NewRequest(http.MethodGet, endpointURL.String(), nil)
+	if err != nil {
+		return fmt.Errorf("Failed to create a new service request: %v", err)
+	}
+
+	if e.credentials != nil {
+		serviceReq.SetBasicAuth(e.credentials.User, e.credentials.Password)
+	}
+
 	go func() {
 		for {
 			select {
 			case <-ticker.C:
-
-				get, _ := http.NewRequest(http.MethodGet, endpointURL.String()+"system/functions", nil)
-				if e.credentials != nil {
-					get.SetBasicAuth(e.credentials.User, e.credentials.Password)
-				}
-
 				services := []types.FunctionStatus{}
-				res, err := proxyClient.Do(get)
+				res, err := proxyClient.Do(serviceReq)
```

Comment:
I don't think we should be re-using the same HTTP request object, that seems wrong.

Why was that change made?

Reply:
You mentioned in one of the previous comments that the request can be cached. From what I understand, a http request can be cached if the body is nil, which is the case here. I will make the change back to original if you think it's safer that way.